include/targets/generic/kernel_host.h

```diff
@@ -6,12 +6,14 @@ namespace quda
   template <template <typename> class Functor, typename Arg> void Kernel1D_host(const Arg &arg)
   {
     Functor<Arg> f(const_cast<Arg &>(arg));
+#pragma omp parallel for
     for (int i = 0; i < static_cast<int>(arg.threads.x); i++) { f(i); }
   }
 
   template <template <typename> class Functor, typename Arg> void Kernel2D_host(const Arg &arg)
   {
     Functor<Arg> f(const_cast<Arg &>(arg));
+#pragma omp parallel for
```

Comment:
AFAIK this will only parallelize the outer loop. To parallelize both loops you can add the clause `collapse(2)` i.e.
```
#pragma omp parallel for collapse(2)
for(int i=0, ... ) {
 for(int j=0, ...) {
 }
 }
 ```

Reply:
Yep, that's indeed the case.  This is very much a first pass at host parallelization and there's a bunch of things one can do to improve this, such as collapsing as you note.  We also don't know whether it's best to have the `i` or `j` dimension as the slowest running one.  The reason I left it without collapse(2) is that the `j` dimension is more often mapped to a dimension which has temporal locality, so naively may exhibit good L1 reuse.

What I eventually plan to do here is to add auto tuner support for the host kernels which would autotune the order of the loops, e.g., which dimension runs outer most, and the degree of parallelization.

So for now, I would suggest leaving it as it is.